security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+    - [Divergence between RISCV.sol and Asterisc](#divergence-between-riscvsol-and-asterisc)
+    - [Generic items we need to take into account:](#generic-items-we-need-to-take-into-account)
+  - [Audit Requirements](#audit-requirements)
+  - [Appendix](#appendix)
+    - [Appendix A: op-challenger](#appendix-a-op-challenger)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Stage 1.4: Kona and Asterisc FMA (Failure Modes and Recovery Path Analysis)
+
+| | |
+|--------|--------------|
+| Author | Andreas Bigger |
+| Created at | 2024-10-17 |
+| Initial Reviewers | Matt Solomon, Ethnical |
+| Need Approval From | Ben Clabby, Mark Tyneway |
+| Status | In Review |
+
+
+## Introduction
+
+This document covers the initial deployment of [the Asterisc Fault Proof VM][asterisc] with [the Kona Fault Proof Program][kona].
+
+Together, Kona and Asterisc form an alternative proof stack to the [op-program] and [cannon].
+A secondary dispute game will be created that uses kona and asterisc.
+The [op-challenger] supports playing _both_ game types - `op-program` + `cannon` as well as `kona` + `asterisc`.
+
+Stage 1.4 ships these two new net components: [Kona Fault Proof Program][kona] and the [Asterisc Fault Proof VM][asterisc].
+Both of these introduce points of failure that are covered in the Failure Modes and Recovery Paths below.
+
+Stage 1.4 also requires the integration of the Kona + Asterisc Dispute Game into the [op-challenger] offchain
+fault proof component. The Failure Modes and Recovery Paths below detail this as well.
+
+Below are references for this project:
+
+- ["Stage 1.4" internal project document](https://www.notion.so/oplabs/Stage-1-4-Partway-a6f57ad777b148dda01488f2646cff17)
+- [Kona Documentation](https://anton-rs.github.io/kona/).
+- [Kona Repository](https://github.com/anton-rs/kona).
+- [Asterisc Repository](https://github.com/ethereum-optimism/asterisc)
+
+[kona]: https://github.com/anton-rs/kona
+[asterisc]: https://github.com/ethereum-optimism/asterisc
+[cannon]: https://github.com/ethereum-optimism/optimism/tree/develop/cannon
+[op-program]: https://github.com/ethereum-optimism/optimism/tree/develop/op-program
+
+
+## Failure Modes and Recovery Paths
+
+### Kona and Asterisc Hot Path Failure
+
+- **Description:**
+  The `op-program` + `cannon` dispute game breaks and the `kona` + `asterisc` fault dispute game is placed in the hot path.
+  If `kona` + `asterisc` also fail, the permissioned fallback will need to be activated.
+
+- **Risk Assessment:**
+  Low likelihood, high impact.
+  Low likelihood since `op-program` + `cannon` are already running in production and have meaningful improvements on the way that will increase resilience.
+  If `kona` + `asterisc` fails while in the hotpath, the impact is high since the permissioned fallback will need to be triggered.
+  This engages the security council, requires comms, and causes the chain to fallback to a weaker security model.
```

Comment:
A small clarification here is that the deputy guardian (foundation) can change the respected game type without involving the security council, so they are not necessarily involved